repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationPropagationTasks.java

```diff
@@ -38,11 +40,20 @@ public Add(AtlasTask task, AtlasGraph graph, EntityGraphMapper entityGraphMapper
         @Override
         protected void run(Map<String, Object> parameters) throws AtlasBaseException {
             String entityGuid               = (String) parameters.get(PARAM_ENTITY_GUID);
-            String classificationVertexId   = (String) parameters.get(PARAM_CLASSIFICATION_VERTEX_ID);
-            String relationshipGuid         = (String) parameters.get(PARAM_RELATIONSHIP_GUID);
+            String tagTypeName              = getTaskDef().getTagTypeName();
+
             Boolean previousRestrictPropagationThroughLineage = (Boolean) parameters.get(PARAM_PREVIOUS_CLASSIFICATION_RESTRICT_PROPAGATE_THROUGH_LINEAGE);
             Boolean previousRestrictPropagationThroughHierarchy = (Boolean) parameters.get(PARAM_PREVIOUS_CLASSIFICATION_RESTRICT_PROPAGATE_THROUGH_HIERARCHY);
-            entityGraphMapper.propagateClassification(entityGuid, classificationVertexId, relationshipGuid, previousRestrictPropagationThroughLineage,previousRestrictPropagationThroughHierarchy);
+
+            if (JANUS_OPTIMISATION_ENABLED != null && JANUS_OPTIMISATION_ENABLED) {
```

Comment:
I see multiple source of truth for flag JANUS_OPTIMISATION_ENABLED
For tasks it is in ClassificationTask initialized in Constructor.
For other instances it is always queries in Redis on-demand.

Let us have only one source for this flag & that too can be initialized only once when server starts.
I am assuming that we will restart Atlas after we set flag to TRUE after migration

Reply:
"For tasks it is in ClassificationTask initialized in Constructor"

Will need to check this point though^